### DISCUSSION TITLE
Audio files often fail on Safari (but not other browsers)

### DISCUSSION THREAD

**Discussion by chrisheffner** [jspsych/jsPsych#2823] [General]
Hi all,

I have an experiment I've been working on for a while that involves participants performing an AX discrimination task, learning to categorize the items that they heard, and then performing a second AX discrimination task. All of the items being presented are sound files, though in the categorization block they are often followed by images. Over the course of the first AX block there start emerging large pauses between the presentation of each sound file in the AX task, and eventually the second sound file starts being skipped. Further into the AX block, the experiment goes blank and the experiment cannot be advanced.

I've deployed the project using both JATOS and Google App Engine; it works successfully on Firefox and Chrome for both, but does not work on Safari. The issues with Safari (and the lack of issues on Chrome/Firefox) have been replicated on multiple machines. No error messages come up on the console. I'm a bit stumped; I'm not sure what to do without any error messages appearing. I may just forbid users from using Safari.

Does anyone know of some sort of Safari-based workaround? Happy to share some code; I'm not entirely sure what's relevant in this case, since it works fine in other browsers, so let me know what you'd like to see.

Thanks in advance!

**Comment by jodeleeuw** [comment]
Hey @chrisheffner,

Just to clarify: the audio files are played at first, but then as the experiment progresses there's a delay and then eventually a crash of some kind?

**Comment by chrisheffner** [comment]
Yes! I'm making up trial numbers here, and these are just the impressions I have from lab members trying on Safari (I'm not a Mac user), but it looks something like this:


  1.  Trials 1-30: some normal AX trials, some with a large gap between the A and X trials
  2.  Trials 31-60: mostly large A and X gap trials, with some X trials not being played at all
  3.  Trials 60-75: most X trials are not being played
  4.  Trial 76: nothing plays and the screen is completely white. Participant cannot advance to the next trial

Chris
